sycl/source/detail/queue_impl.cpp

```diff
@@ -217,6 +217,11 @@ void queue_impl::wait(const detail::code_location &CodeLoc) {
     std::lock_guard<mutex_class> Lock(MMutex);
     Events = std::move(MEventsWeak);
     USMEvents = std::move(MEventsShared);
+
+    // moved objects are in "valid but unspecified state". It's okay to only
+    // clear the vector. Though, we'll recostruct it hust to be sure.
```

Comment:
Good catch! LGTM.

I'm bad at wording comments, but maybe  "moved objects are in 'valid but unspecified state'.  For safety, we reconstruct empty vectors."